addons/stock_account/models/product.py

```diff
@@ -223,15 +223,18 @@ def _change_standard_price(self, new_price):
             quantity_svl = product.sudo().quantity_svl
             if float_compare(quantity_svl, 0.0, precision_rounding=product.uom_id.rounding) <= 0:
                 continue
-            diff = new_price - product.standard_price
+            precision = self.env['decimal.precision'].precision_get('Product Price')
+            rounded_new_price = fields.Float.round(new_price, precision_digits=precision)
```

Comment:
you could use float_repr for that. Or even value_to_html from ir.qweb.field.float to also have the correct format from the user lang

Reply:
I would agree if it was only used for the description field. Maybe my commit's description is not clear enough ^^

But the `new_price` is also used to compute the `diff` which is then used to compute the `value`. And if we don't round at that point, there will be slight mistakes. Mistakes that can have an impact for great quantities.